src/main/java/org/opentestsystem/ap/tms/packager/manager/FixedFormPackagerManager.java

```
@@ -0,0 +1,107 @@
+            throw new RuntimeException(String.format("An exception occurred while creating the file %s", outputFileFullPath), e);
+        }
+
+        return Optional.of(new File(outputFileFullPath));
+    }
+
+    private static String getBankKey(final TestPackageSheet sheet) {
+        return sheet.getString("BankKey");
+    }
+
+    private static String[] getItemIds(final String bankKey, final TestPackageSheet sheet) {
+//        final String[] itemIds = sheet.getStrings("ItemId");
+//
+//        // Item IDs in the input file do not include the bank key - but the items are keyed by the entire item key (which is
+//        // prefixed by the bank key) in gitlab
+//        for (int i = 0; i < itemIds.length; i++) {
+//            itemIds[i] = String.format("%s-%s", bankKey, itemIds[i]);
+//        }
+//
+//        return itemIds;
+        return sheet.getStrings("ItemId");
+    }
+
+    private TestPackageWorkbook createTestPackageWorkbook(final String inputSpreadsheetPath) {
+        try {
+            return new TestPackageWorkbook(new XSSFWorkbook(new FileInputStream(inputSpreadsheetPath)));
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading test package spreadsheet.", e);
+        }
+    }
+
+    private void createAndValidateTestPackageFile(final String outputFilePath, final TestPackage testPackage,
+                                                  final boolean debug) throws IOException {
+        final File testPackageFile = new File(outputFilePath);
+        xmlMapper.writeValue(testPackageFile, testPackage);
+
+        try (FileInputStream fis = new FileInputStream(testPackageFile)) {
+            final StreamSource xmlFile = new StreamSource(fis);
+            schemaValidator.validate(xmlFile);
+            testPackageFile.createNewFile();
+        } catch (SAXException e) {
+            System.out.println("Error during XSD validation of the test package " + testPackage.getId());
```

I feel like this should throw an Exception that will bubble and return to the user.  I'm not sure if this should be a ValidationException or a SystemException.  I think if this is the part that validates the things entered into the spreadsheet are correct I think ValidationException is appropriate.

I also think we should log the Exception.  Defer to you at what level.

---

Changed to SystemException